q2_assembly/quast/tests/test_quast.py

```diff
@@ -757,7 +753,9 @@ def test_create_tabular_results_in_combined_subdir(self, p1, p2):
         p2.assert_called_once_with(report_path, sep="\t", header=0)
         p1.assert_called_once_with(mock_df, [1000, 5000, 25000, 50000])
 
-    def test_evaluate_contigs_pipeline(self):
+    @parameterized.expand([(True, True), (False, True), (False, False)])
+    @patch("warnings.warn")
```

Comment:
I must say, that only now that you removed this itertools import I see what's happening in this test. The bad news is - all these cases you are testing here should be _separate_ test cases. This way it is very hard to understand what is actually happening in your test with all these nested if statements. Unit tests should be only as complex as really necessary - please split this into three different, smaller tests (maybe there's something else you can do, though, to avoid code repetition).